test-case/verify-bootsequence.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+##
+## Case Name: verify-bootsequence.sh
+## Preconditions:
+##    UCMv2 is installed in /usr/share/alsa/ucm2 folder
+## Description:
+##    Run alsactl init to test the initialization settings are correct or not
+## Case step:
+##    1. check the alsa-lib, alsa-utils version
+##    2. save the original amixer settings
+##    3. run alsactl init
+##    4. restore the original amixer settings
+## Expect result:
+##    alsactl init runs successfully
+##
+
+set -e
+
+# shellcheck source=case-lib/lib.sh
+source "$(dirname "${BASH_SOURCE[0]}")"/../case-lib/lib.sh
+# shellcheck source=case-lib/user.sh
+source "$(dirname "${BASH_SOURCE[0]}")"/../case-lib/user.sh
+
+# BootSequence is introduced in alsa-utils 1.2.2-16
+ALSAUTILS_VER=1.2.2-16
+# BootSequence is introduced in alsa-lib 1.2.2-44
+ALSALIB_VER=1.2.2-44
+# BootSequence is introduced in alsa-ucm-conf 1.2.3-15
+ALSAUCM_VER=1.2.3-15
+
+func_opt_parse_option "$@"
+
+main()
+{
+    local tmp
+
+    func_check_version "libasound2" "$ALSALIB_VER"
+    func_check_version "alsa-utils" "$ALSAUTILS_VER"
+    func_version_ge "alsa-ucm-conf" "$ALSAUCM_VER" || dlogi "alsa-ucm-conf version is too low"
+
+    # alsactl init only applies to SOF card
+    [[ -n "$SOFCARD" ]] || die "No SOF card"
+    cardname=$(sof-dump-status.py -s "$SOFCARD") || exit 1
+    [[ -n "$cardname" ]] || die "Failed to get card $SOFCARD short name"
+
+    tmp=$(mktemp -d) || die "Failed to create tmp folder."
+    cd "$tmp"
+    dlogi "Checking the card $cardname UCM2 initial settings"
```

Comment:
This log message is misleading because it looks like the test does something UCM-specific but it does not. Change to something like:

> dlogi "Indirectly checking the card $cardname UCM2 initial settings (if any) with alsactl store"

Reply:
> This log message is misleading because it looks like the test does something UCM-specific but it does not. Change to something like:
> 
> > dlogi "Indirectly checking the card $cardname UCM2 initial settings (if any) with alsactl store"

@marc-hb Yes. Your suggested dlogi is more reasonable. I will use it expect "with alsactl init" not "with alsactl store"

>> two version check functions that does not achieve the goal of having one dpkg-independent functions

Yes. Currently I don't have other better solutions to check the version. I tried commands such as `aplay --version`, but the version information is not enough for us. And what's more, I can't get the alsa-lib and alsa-ucm-conf version with such command. 
If we found a better solution, I will update it.

Reply:
I mean this (untested)
```shell
dpkg_version_ge()
{
    ver=$(dpkg -l "$1" |grep "$1" | awk '{print $3}')
    [[ -z "$ver" ]] && die "failed to get $1 version"
    dlogi "find $1 version  $ver"
    version_ge "$1" "$2"
}
 
version_ge()
{   
    [ "$(printf '%s\n' "$2" "$ver" | sort -V | head -n1)" = "$2" ]
}
```